libr/util/p_format.c

```diff
@@ -1079,22 +1089,64 @@ int r_print_format_struct_size(const char *f, RPrint *p, int mode) {
 			{
 			const char *format = NULL;
 			char *endname = NULL, *structname = NULL;
+			char tmp = 0;
 			structname = strdup (r_str_word_get0 (args, idx));
 			if (*structname == '(') {
-				endname = strchr (structname, ')');
+				endname = strrchr (structname, ')');
 			} else {
 				eprintf ("Struct name missing (%s)\n", structname);
 				free (structname);
 				break;
 			}
 			if (endname) *endname = '\0';
-			format = r_strht_get (p->formats, structname + 1);
-			free (structname);
+			format = strchr (structname, ' ');
+			if (format) {
+				tmp = *format;
+				while (tmp == ' ') {
+					format++;
+					tmp = *format;
+				}
+			} else {
+				format = r_strht_get (p->formats, structname + 1);
+			}
 			size += tabsize * r_print_format_struct_size (format, p, mode);
+			free (structname);
 			}
 			break;
-			// TODO continue list
+		case '{':
+			while (fmt[i] != '}') {
+				i++;
+			}
+			i++;
+			idx--;
+			break;
+		case '}':
+			free (o);
+			free (args);
+			return -1;
+			break;
+		case '+':
+		case 'e':
+			idx--;
+			break;
+		case 'p':
+			if (fmt[i+1] == '2') {
```

Comment:
use braces in oneliners, this is how we decided to code after 1 year of discussions :P and the code style is slowly changing towards this



> On 17 Aug 2016, at 22:28, Jaime Peñalba <notifications@github.com> wrote:
> 
> In libr/util/p_format.c <https://github.com/radare/radare2/pull/5547#discussion_r75200353>:
> 
> > +				i++;
> > +			}
> > +			i++;
> > +			idx--;
> > +			break;
> > +		case '}':
> > +			free (o);
> > +			free (args);
> > +			return -1;
> > +			break;
> > +		case '+':
> > +		case 'e':
> > +			idx--;
> > +			break;
> > +		case 'p':
> > +			if (fmt[i+1] == '2') {
> Do not use braces for one line statements
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/radare/radare2/pull/5547/files/96365d8ce8d4f28f3ddef148e49729f044df1a20#r75200353>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA3-lrt9EwnYDke9oO8mdUuGE8w6N3RNks5qg27ggaJpZM4Jm2Tj>.
>